Merged commits, backported to 2.6.15 and modified not to require nonfree for
nvenc:

commit f845288d3d1f5bc8f27eb76a4aacd111ca0811bc
Author: eumagga0x2a <eumagga0x2a@users.noreply.github.com>
Date:   Sat Dec 17 23:28:39 2016 +0100

    [build/cmake] Fix NVENC detection and ffmpeg configure options

 cmake/admCheckNvEnc.cmake         | 5 +++--
 cmake/admFFmpegBuild_native.cmake | 1 +
 2 files changed, 4 insertions(+), 2 deletions(-)


commit 25d94f28cbae1a25e7dcdcce106f5bc5c4b637df
Author: eumagga0x2a <eumagga0x2a@users.noreply.github.com>
Date:   Sat Dec 17 23:56:24 2016 +0100

    [build/cmake] Fix the status message from the previous commit

 cmake/admCheckNvEnc.cmake | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)


commit f27e9e53929fb08852b767b66b035a6ccd4cd404
Author: mean <fixounet@free.fr>
Date:   Sun Dec 18 08:25:03 2016 +0100

    [build/core] update ffmpeg to 3.0.5

 avidemux_core/ffmpeg_package/ffmpeg-3.0.3.tar.bz2 | Bin 8885942 -> 0 bytes
 avidemux_core/ffmpeg_package/ffmpeg-3.0.5.tar.bz2 | Bin 0 -> 8890675 bytes
 cmake/admFFmpegBuild_helpers.cmake                |   2 +-
 3 files changed, 1 insertion(+), 1 deletion(-)

diff -rpU 3 a/cmake/admCheckNvEnc.cmake b/cmake/admCheckNvEnc.cmake
--- a/cmake/admCheckNvEnc.cmake	2016-11-19 08:58:39.000000000 +0100
+++ b/cmake/admCheckNvEnc.cmake	2016-12-20 22:32:38.287770324 +0100
@@ -6,10 +6,10 @@ MACRO(checkNvEnc)
 		MESSAGE(STATUS "*****************")
 
 		IF (NVENC)
-                        FIND_PATH(NVENC_INCLUDE_DIR nvEncodeAPI.h 
-		      	PATHS /usr/include/x86_64-linux-gnu) # Needed for 64 bits linux
+                        FIND_PATH(NVENC_INCLUDE_DIR nvEncodeAPI.h
+                        PATHS /usr/include /usr/include/nvenc /usr/include/x86_64-linux-gnu) # Needed for 64 bits linux
                         IF(NVENC_INCLUDE_DIR)
-				MESSAGE(STATUS " nvenc header Found ")
+				MESSAGE(STATUS " nvenc header Found in ${NVENC_INCLUDE_DIR}")
                                 SET(USE_NVENC True)
                                 SET(NVENC_FOUND 1)
                         ELSE(NVENC_INCLUDE_DIR)
diff -rpU 3 a/cmake/admFFmpegBuild.cmake b/cmake/admFFmpegBuild.cmake
--- a/cmake/admFFmpegBuild.cmake	2016-11-19 08:58:39.000000000 +0100
+++ b/cmake/admFFmpegBuild.cmake	2016-12-20 22:49:20.968852129 +0100
@@ -9,7 +9,7 @@ ENDMACRO (xadd)
 
 option(FF_INHERIT_BUILD_ENV "" ON)
 
-set(FFMPEG_VERSION "3.0.3")
+set(FFMPEG_VERSION "3.0.5")
 set(FFMPEG_ROOT_DIR "${AVIDEMUX_TOP_SOURCE_DIR}/avidemux_core/ffmpeg_package")
 set(FFMPEG_PATCH_DIR  "${FFMPEG_ROOT_DIR}/patches/")
 set(FFMPEG_SOURCE_ARCHIVE "ffmpeg-${FFMPEG_VERSION}.tar.bz2")
@@ -45,8 +45,8 @@ xadd("--disable-libx265")
 xadd("--disable-libx264")
 IF(USE_NVENC)
    SET(FFMPEG_ENCODERS ${FFMPEG_ENCODERS} nvenc)
-   xadd("--enable-nonfree")
    xadd("--enable-nvenc")
+   xadd("--extra-cflags=-I${NVENC_INCLUDE_DIR}")
    set(FFMPEG_ENCODERS  ${FFMPEG_ENCODERS} nvenc_h264 nvenc_hevc)
 ENDIF(USE_NVENC)
 
@@ -93,6 +93,8 @@ if (FFMPEG_PERFORM_PATCH)
 	if (UNIX )
 			MESSAGE(STATUS "Patching Linux common.mak")
 			patch_file("${FFMPEG_SOURCE_DIR}" "${FFMPEG_PATCH_DIR}/common.mak.diff") 
+			MESSAGE(STATUS "Patching configure not to require nonfree for nvenc")
+			patch_file_p1("${FFMPEG_SOURCE_DIR}" "${FFMPEG_PATCH_DIR}/ffmpeg-3.0.5-do-not-require-nonfree-for-nvenc.diff")
 	endif (UNIX )
         IF(WIN32)
                 IF("${CROSS_C_COMPILER}" STREQUAL "clang")
